cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -640,8 +640,8 @@ private void executeAll(Iterator<WorkflowNode> iterator, ApplicationSpecificatio
           workflowContext.setState(new ProgramState(ProgramStatus.KILLED, rootCause.getMessage()));
           break;
         }
-        workflowContext.setState(new ProgramState(ProgramStatus.FAILED, rootCause.getMessage()));
-        throw Throwables.propagate(rootCause);
+        workflowContext.setState(new ProgramState(ProgramStatus.FAILED, t.getMessage()));
```

Comment:
similar question here, what gets thrown by executeNode? It seems like there is some sort of wrapping going on, since it is looking for an InterruptedException as the root cause. 

Asking since it would be good to also remove extra wrapping if it is happening.

Reply:
ExecuteNode seems to throw any exceptions which happen during the runtime of a pipeline. It recursively calls ExecuteCondition, ExecuteAll, ExecuteFork, ExecuteCustomAction, and ExecuteAction so it could really be any kind of exception.